src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,809 @@
+                     self._port, self._idn, self._version)
+
+        self._offset = {}
+        self._axis_conv_factor = {}
+        self._axis_unit = {}
+
+        # Not to be mistaken with axes which is a simple public view
+        self._axis_map = {}  # axis name -> axis number used by controller
+        axes_def = {}  # axis name -> Axis object
+        position = {}
+        speed = {}
+        accel = {}
+        decel = {}
+
+        for axis_name, axis_par in axes.items():
+            # Unpack axis parameters from the definitions in the YAML
+            try:
+                axis_num = axis_par['number']
+            except KeyError:
+                raise ValueError("Axis %s must have a number to identify it. " % (axis_name,))
+
+            try:
+                axis_range = axis_par['range']
+            except KeyError:
+                logging.info("Axis %s has no range. Assuming (-1, 1)", axis_name)
+                axis_range = (-1, 1)
+
+            try:
+                axis_unit = axis_par['unit']
+            except KeyError:
+                logging.info("Axis %s has no unit. Assuming m", axis_name)
+                axis_unit = "m"
+
+            if axis_unit not in ('m', 's'):
+                raise ValueError("Invalid axis unit. Should be m or s.")
+
+            try:
+                conv_factor = float(axis_par['conv_factor'])
+            except KeyError:
+                logging.info("Axis %s has no conversion factor. Assuming 1000 (m to mm)", axis_name)
+                conv_factor = 1000.0
+
+            self._axis_map[axis_name] = axis_num
+            self._axis_conv_factor[axis_num] = conv_factor
+            self._axis_unit[axis_num] = axis_unit
+            speed[axis_name] = self.GetSpeed(axis_num)
+            accel[axis_name] = self.GetAcceleration(axis_num)
+            decel[axis_name] = self.GetDeceleration(axis_num)
+
+            # Force millimetres for consistency as the internal unit.
+            self.SetAxisUnit(axis_num, "mm")
+            # initialize each motor
+            self.MotorOn(axis_num)
+
+            ad = model.Axis(canAbs=False, unit=axis_unit, range=axis_range)
+            axes_def[axis_name] = ad
+
+        model.Actuator.__init__(self, name, role, axes=axes_def, **kwargs)
+
+        # Get the position in object coord with the offset applied.
+
+        # RO, as to modify it the client must use .moveRel() or .moveAbs()
+        self.position = model.VigilantAttribute({}, unit=position_display_unit, readonly=True)
+        self._updatePosition()
+
+        self._speed = speed
+        self._accel = accel
+        self._decel = decel
+
+        # set offset due to mounting of components (float)
+        self._metadata[model.MD_POS_COR] = 0.
+
+        # will take care of executing axis move asynchronously
+        self._executor = ParallelThreadPoolExecutor()  # one task at a time
+
+        # Check the error state
+        self.checkError()
```

Comment:
You probably should read all the errors until it runs out of error just after connecting to the device. This way, if there are some errors queued from before, they won't make the whole initialisation fail at the end. 
If you don't do that, checking the error state here is pretty much shooting yourself in the foot: the whole initialisation goes fine, and then you raise an Exception.